ext/crypto/00_crypto.js

```diff
@@ -4326,7 +4326,13 @@
             length,
           });
 
-          return buf.buffer;
+          // 8.
+          if (!length) {
+            return buf.buffer;
+          } else if (buf.buffer.byteLength * 8 < length || length % 8 !== 0) {
+            throw new DOMException("Invalid length", "OperationError");
+          }
+          return buf.buffer.slice(0, length / 8);
```

Comment:
```suggestion
          if (length === null) {
            return buf.buffer;
          }
          
          if (typeof length !== "number" || buf.buffer.byteLength * 8 < length || length % 8 !== 0) {
            throw new DOMException("Invalid length", "OperationError");
          }
          
          return buf.buffer.slice(0, length / 8);
```

Reply:
Deno casts `null` as 0 for the `length` argument for some reason. This leads to returning a 0-sized buffer, which is incorrect regarding the spec.

Reply:
That's a problem. `null` is to be handled, `0` is required to be thrown when provided (this is part of the WPT tests which are, unfortunately, not enabled for ECDH).

Reply:
That's not how I understand the spec. Maybe this detail can be sorted out when the tests are enabled? I don't know the project well enough to understand why this cast occurs.

Reply:
In that case, don't use the casted for the null check. Otherwise you can throw when 0 is passed.

```suggestion
          // 8.
          if (arguments[2] === null) {
            return buf.buffer;
          }
          
          if (typeof length !== "number" || length === 0 || buf.buffer.byteLength * 8 < length || length % 8 !== 0) {
            throw new DOMException("Invalid length", "OperationError");
          }
          
          return buf.buffer.slice(0, length / 8);
```

Reply:
I finally found where the cast happened, so I included your change.

Reply:
👍

Reply:
>  why this cast occurs.

Its done in `serde_v8`. The codebase has adapted to the behaviour by using `Option<u32>`